interfaces/cython/cantera/preconditioners.pyx

```diff
@@ -13,6 +13,26 @@ cdef class PreconditionerBase:
     def __cinit__(self, *args, **kwargs):
         self.pbase = newPreconditioner(stringify(self.precon_type))
 
+    property side:
+        """
+        Get/Set the side of the system matrix where the preconditioner is applied.
+        Options are "none", "left", "right", or "both". Not all options are supported
+        by all solver types.
+        """
+        def __get__(self):
+            cdef int side = <int>self.pbase.get().preconditionerSide()
+            if side == 0:
```

Comment:
See comment for C++ side: the Python API has to convert what appear to be magic numbers (the `enum` type in C++ is clearly not an issue).